include/wx/charts/wxpiechart.h

```diff
@@ -27,19 +27,18 @@
 
 #include "wxdoughnutandpiechartbase.h"
 #include "wxpiechartoptions.h"
+#include "wxchartobservers.h"
 
 /// Data for the wxPieChartCtrl control.
-class wxPieChartData
+class wxPieChartData : public wxChartObservableValue<wxVector<wxChartSliceData>>
 {
 public:
     wxPieChartData();
 
-    void AppendSlice(const wxChartSliceData &slice);
-
     const wxVector<wxChartSliceData>& GetSlices() const;
-
-private:
-    wxVector<wxChartSliceData> m_slices;
+    void AppendSlice(const wxChartSliceData &slice);
+    void UpdateData(const wxVector<wxChartSliceData> &data);
```

Comment:
Last comment and I'll merge the review. Can we rename these UpdateSlice and AddSlices, just to keep the naming consistent.